### ISSUE TITLE
voiding addon contract causes "postive balance remains" error

### ISSUE THREAD

**Issue by kbower** [kbower/tickettest1#1622]
_Imported from trac ticket 1622.
Created by **rneville, sal**
Opened in trac: Apr 26, 2012, 2:20:17 PM
Last modified in trac: May 7, 2012, 12:16:45 PM_
***
_by rneville, sal:_
I got the following error when trying to void SO# 0424259DI76 on train:
 
Somehow a positive balance remains when voiding this contract.
 
The difference between the one I sent earlier that actually let me void and this one is that this one is an add-on contract.
 
Sal

This is because:
```python
        for order in self.currentorders:
            saletran = order.saletransaction
            saletran.paymenttypeid = None
            saletran.artype = ArTran.ARTYPE_ORDER
            saletran.contractid = None
            order.financedamount = Decimal('0')
            
        #go through deposits made against this contractid and "revert" them
        #to be deposits on the order instead since we're about to void this
        #contract (rather than leave a negative balance on the old contract)
        for tran in self._get_dep_list():
            tran.orderid = saletran.orderid
            tran.contractid = None
            tran.artype = ArTran.ARTYPE_ORDER

        self.voided = u'Y'
        
        for tran in self._get_debit_transaction_list():
            self._assembleartran(self._invert_transaction_code(tran.transactioncodeid), tran.amount)
        
        self.expire('transactions') #forces re-eval of current balance given all the changes we just made

        remainingbalance = self.currentbalance
        if remainingbalance < Decimal('0'): 
            remainingbalance = abs(remainingbalance)
        elif remainingbalance > Decimal('0'):
            raise RError("Somehow a positive balance remains when voiding this contract")
```
Is happening *before* the original contract is being dealt with right below that part of the code...so the DTD is still sitting there on the contract at that point and will fail every time.

**Comment by kbower** [comment]
_**rneville** trac comment on May 7, 2012, 12:16:45 PM:_

(Changeset [[2631]](../commit/85fb241046f1cfdbfea20dca35e71603076c6c3d))
 * fixes #1622 voiding addon contract causes "postive balance remains" error

**Comment by kbower** [comment]
_**rneville** trac status change on May 7, 2012, 12:16:45 PM:_ **closed**
